subsys/mpsl/mpsl_init.c

```diff
@@ -54,6 +54,12 @@ static int mpsl_lib_init(struct device *dev)
 		return err;
 	}
 
+	IRQ_DIRECT_CONNECT(TIMER0_IRQn, MPSL_HIGH_IRQ_PRIORITY,
+		    MPSL_IRQ_TIMER0_Handler, 0);
+	IRQ_DIRECT_CONNECT(RTC0_IRQn, MPSL_HIGH_IRQ_PRIORITY,
+		    MPSL_IRQ_RTC0_Handler, 0);
+	IRQ_DIRECT_CONNECT(RADIO_IRQn, MPSL_HIGH_IRQ_PRIORITY,
+		    MPSL_IRQ_RADIO_Handler, 0);
```

Comment:
How did MPSL work before this?

Reply:
The MPSL clock driver worked without this. Before working on this sample I was checking if using the MPSL clock driver in zephyr bluetooth examples worked. To make it work there, the IRQs could not be connected to the MPSL handlers as they were already connected. So when making this timeslot sample I initially placed the IRQ connects in the sample. To keep it flexible. But now I moved it to mpsl_init.c.

Reply:
@trond-snekvik MPSL is not being used by the BLE controller yet. So this change was required to make timeslot work.

@sofu1  should we maybe add the ZERO_LATENCY option as well? It would prevent the ISRs from being blocked by irq_lock.

Reply:
I see, thanks!